allauth/account/forms.py

```diff
@@ -271,7 +271,7 @@ def clean_email(self):
         value = self.cleaned_data["email"]
         value = get_adapter().clean_email(value)
         if value and app_settings.UNIQUE_EMAIL:
-            get_adapter().validate_unique_email(value)
+            value = get_adapter().validate_unique_email(value)
```

Comment:
Realized someone may want to do something nuts with validate_unique_email and return something different than the original value for maybe a strange use case, so setting the value to what returns (just returns the original email if valid by default) just to be safe.